### PR TITLE
Crypto wallet UI android

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves https://github.com/brave/brave-browser/issues/16367

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/crypto_wallet/util/Utils.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                    "(?=.*[0-9])" +         //at least 1 digit
+                    "(?=.*[a-zA-Z])" +      //any letter
+                    "(?=.*[$&+,:;=?@#|'<>.^*()%!-])" +    //at least 1 special character
+                    "(?=\\S+$)" +           //no white spaces
+                    ".{7,}" +               //at least 7 characters
+                    "$");
+
+    // public static final List<String> recoveryPhrases =
+    //         new ArrayList<>(Arrays.asList("Tomato", "Green", "Velvet", "Span", "Celery", "Atoms",
+    //                 "Parent", "Stop", "Bowl", "Wishful", "Stone", "Exercise"));
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/crypto_wallet/util/Utils.java:33]: we don't need that

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Comment by deeppandya** [comment]
This is not a complete feature. I will be creating follow up issues and working on it.

File: build/android/bytecode/java/org/brave/bytecode/BraveStandardProtectionSettingsFragmentClassAdapter.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+            "org/chromium/chrome/browser/safe_browsing/settings/StandardProtectionSettingsFragment";
+    static String sBraveStandardProtectionSettingsFragmentClassName =
+            "org/chromium/chrome/browser/safe_browsing/settings/BraveStandardProtectionSettingsFragment";
+
+    public BraveStandardProtectionSettingsFragmentClassAdapter(ClassVisitor visitor) {
+        super(visitor);
+        makePublicMethod(sStandardProtectionSettingsFragmentClassName,
+                "updateLeakDetectionAndExtendedReportingPreferences");
+        changeMethodOwner(sStandardProtectionSettingsFragmentClassName,
+                "updateLeakDetectionAndExtendedReportingPreferences",
```
Comment (samartnik) [inline build/android/bytecode/java/org/brave/bytecode/BraveStandardProtectionSettingsFragmentClassAdapter.java:21]: I don't see `updateLeakDetectionAndExtendedReportingPreferences`method in `BraveStandardProtectionSettingsFragment`. What is this change about?

**Review by samartnik — CHANGES_REQUESTED** [review]

File: android/java/org/chromium/chrome/browser/crypto_wallet/util/SmoothLineChartEquallySpaced.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+        mPath = new Path();
+    }
+
+    public void setData(float[] values) {
+        mValues = values;
+
+        if (values != null && values.length > 0) {
+            mMaxY = values[0];
+            // mMinY = values[0].y;
```
Comment (samartnik) [inline android/java/org/chromium/chrome/browser/crypto_wallet/util/SmoothLineChartEquallySpaced.java:66]: Are these comments left overs?

**Review by samartnik — APPROVED** [review]

File: chromium_src/chrome/browser/flags/android/chrome_feature_list.cc [hunk 4]
```diff
@@ -4,6 +4,7 @@
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include "brave/common/brave_features.h"
+#include "brave/components/brave_wallet/common/features.h"
```
Comment (mkarolin) [inline chromium_src/chrome/browser/flags/android/chrome_feature_list.cc:7]: Should the additions in there be under `#if BUILDFLAG(BRAVE_WALLET_ENABLED)`?

Comment (mkarolin) [reply chromium_src/chrome/browser/flags/android/chrome_feature_list.cc:8]: Also, needs `#include "brave/components/brave_wallet/common/buildflags/buildflags.h"` now before using `#if BUILDFLAG(BRAVE_WALLET_ENABLED)`.

The `#if BUILDFLAG(BRAVE_WALLET_ENABLED)` include typically should go underneath other includes separated from them by an empty line.

Comment (SergeyZhukovsky) [reply chromium_src/chrome/browser/flags/android/chrome_feature_list.cc]: yeah, my bad, missed that and forgot to try to build it locally

**Review by bbondy — APPROVED** [review]
approving to bypass codeowner

**Review by mkarolin — APPROVED** [review]
`chromium_src` change LGTM

File: browser/brave_wallet/BUILD.gn [hunk 5]
```diff
@@ -4,6 +4,8 @@ import("//extensions/buildflags/buildflags.gni")
 assert(brave_wallet_enabled)
 
 source_set("brave_wallet") {
+  check_includes = false
```
Comment (bridiver) [inline browser/brave_wallet/BUILD.gn:7]: This is not ok and needs to be fixed asap. `check_includes = false` should never be added to new targets. @mariospr has been working hard to get rid of these and we should never add new ones

**Pull request merged**